Parser/tokenizer.c

```diff
@@ -2529,8 +2552,9 @@ tok_get_fstring_mode(struct tok_state *tok, tokenizer_mode* current_tok, struct
             if (peek != '{' || in_format_spec) {
                 tok_backup(tok, peek);
                 tok_backup(tok, c);
-                current_tok->bracket_mark[++current_tok->bracket_mark_index] = current_tok->bracket_stack;
-                tok->tok_mode_stack[tok->tok_mode_stack_index].kind = TOK_REGULAR_MODE;
+                current_tok->bracket_mark_index++;
+                *TOK_GET_BRACKET_MARK(current_tok) = current_tok->bracket_stack;
+                TOK_GET_MODE(tok)->kind = TOK_REGULAR_MODE;
```

Comment:
`->` has higher precedence over `&`. Paren the macro definition in no debug mode.